src/swsssdk/dbconnector.py

```diff
@@ -9,182 +9,324 @@
+        'sonic_db_file_path' is explicitly passed as parameter.
         """
         if SonicDBConfig._sonic_db_config_init == True:
             return
 
         try:
-            if os.path.isfile(sonic_db_file_path) == False:
-                msg = "'{}' is not found, it is not expected in production devices!!".format(sonic_db_file_path)
-                logger.warning(msg)
-                sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
-            with open(sonic_db_file_path, "r") as read_file:
-                SonicDBConfig._sonic_db_config = json.load(read_file)
+            if os.path.isfile(sonic_db_file_path) ==  True:
+                with open(sonic_db_file_path, "r") as read_file:
+                    all_dbs = json.load(read_file)
+                    for entry in all_dbs['INCLUDES']:
+                        if 'namespace' not in entry.keys():
+                            namespace = 'default'
+                        else: 
+                            namespace = entry['namespace']
+
+                        db_include_file = SonicDBConfig.SONIC_DB_BASE_DIR+entry['include']
+                        SonicDBConfig._sonic_db_config[namespace] = None
+
+                        # Not finding the database_config.json file for the namespace
+                        if os.path.isfile(db_include_file) == False:
+                            msg = "'{}' file is not found !!".format(db_include_file)
+                            logger.warning(msg)
+                            continue
+
+                        with open(db_include_file, "r") as inc_file:
+                            SonicDBConfig._sonic_db_config[namespace] = json.load(inc_file)
+            else:
+                sonic_db_file_path = SonicDBConfig.SONIC_DB_CONFIG_FILE
+                if os.path.isfile(sonic_db_file_path) ==  False:
+                    msg = "'{}' is not found, it is not expected in production devices!!".format(sonic_db_file_path)
+                    logger.warning(msg)
+                    sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
+
+                with open(sonic_db_file_path, "r") as read_file:
+                    SonicDBConfig._sonic_db_config['default'] = json.load(read_file)
+
         except (OSError, IOError):
             msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
             logger.exception(msg)
             raise RuntimeError(msg)
+
         SonicDBConfig._sonic_db_config_init = True
 
     @staticmethod
-    def db_name_validation(db_name):
+    def db_name_validation(db_name, ns_instance):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        if db_name not in SonicDBConfig._sonic_db_config["DATABASES"]:
+        data=SonicDBConfig._sonic_db_config[ns_instance]["DATABASES"]
```

Comment:
>ns_instance [](start = 44, length = 11)

what happens if ns_instance does not exist in _sonic_db_config?

Reply:
I have added a new API namespace_validation() introduced to check the namespace validation similar to instance/db_name.